backend/gce.go

```diff
@@ -930,23 +939,23 @@ func (i *gceInstance) sshConnection(ctx gocontext.Context) (ssh.Connection, erro
 }
 
 func (i *gceInstance) getIP() string {
-	// if instance has no public IP, return first private one
-	if !i.ic.PublicIP {
+	if i.ic.PublicIP && i.ic.PublicIPConnect {
```

Comment:
Is the `PublicIPConnect` bit even needed?  This looks to me like `PublicIP` is enough.